### PR TITLE
feat: allow local file playback of mobile recordings

### PR DESCRIPTION
**Pull request by pauldambra**
fixes local file replay so that it works with exported mobile replay data

pairs with Chianti or https://github.com/PostHog/posthog/pull/19263

Tested by playing back one of each and seeing them succeed

---

And adds fix for progress spinner

NB the spinner spins because of animation and rrweb pauses animations when the player is paused

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts [hunk 1]
```diff
@@ -56,14 +60,14 @@ const parseEncodedSnapshots = async (
...
-            const snapshotData = snapshotLine['data']
+            const snapshotData = isRecordingSnapshot(snapshotLine) ? [snapshotLine] : snapshotLine['data']
 
             return snapshotData.map((d: unknown) => {
                 const snap = withMobileTransformer
                     ? postHogEEModule?.mobileReplay?.transformEventToWeb(d) || (d as eventWithTime)
                     : (d as eventWithTime)
                 return {
-                    windowId: snapshotLine['window_id'],
+                    windowId: snapshotLine['window_id'] || snapshotLine['windowId'],
```
Comment (marandaneto) [inline frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:70]: shouldn't be only one way since it's our protocol?

Comment (pauldambra) [reply frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:70]: 🤔  it's this method that's normalising the windowId before export

So, the stored value has `window_id` but then this transforms it for the UI, then we export it... 🤔 

I'll do another pass and see how painful that'd be to sort out

Comment (pauldambra) [reply frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:74]: would be a bit of a pain to do a quick version of this...

added a comment instead

Will punt into the future

Comment (pauldambra) [reply frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:74]: (because I want to do the styling changes 😊)

Comment (marandaneto) [reply frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts:74]: If there's a comment to explain why we fallback to the other option, all good.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**